### ISSUE TITLE
Generate OpenCL backend delegate

### ISSUE THREAD

**Issue by aitikgupta** [ValYouW/tflite-dist#10]
The series of steps you followed generates (for android), 3 dynamic libs:

1. libtensorflowlite.so - C++ library
2. libtensorflowlite_c.so - C library
3. libtensorflowlite_gpu_gl.so - The C++ GPU delegate library

However, since I believe 2.2 (I believe) release, TFLite supports 2 backends for android, OpenGL (the one you generate) and one OpenCL.
OpenCL is the default, and if some device doesn't support it, it automatically falls back to OpenGL backend.

The change in the build process should be:
```console
bazel build -c opt --config android_{abi} tensorflow/lite/delegates/gpu:libtensorflowlite_gpu_delegate.so
```
instead of
```console
bazel build -c opt --config android_{abi} tensorflow/lite/delegates/gpu:libtensorflowlite_gpu_gl.so
```

**Comment by ValYouW** [comment]
Hi @aitikgupta thank you! I'll take a look. I have code using OpenGL and I pass it as input SSBO object (so I avoid cpu/gpu hops), I wonder if the api for the new gpu is the same

**Comment by aitikgupta** [comment]
The new API is (almost) same.
"V2" is appended for some public functions like: `TfLiteGpuDelegateV2Create`, `TfLiteGpuDelegateOptionsV2`, etc.
Documentation is updated on their website: https://www.tensorflow.org/lite/performance/gpu_advanced#android_cc

Quoting their [announcement blog](https://blog.tensorflow.org/2020/08/faster-mobile-gpu-inference-with-opencl.html):
>  In fact, the OpenCL backend has been in the TensorFlow repository since mid 2019 and seamlessly integrated through the TFLite GPU delegate v2, so you might be already using it through the delegate's fallback mechanism.

**Comment by ValYouW** [comment]
@aitikgupta Released v2.4.1 with the gpu_delegate, thank you for bringing that up....
